plugins/offboard/offboard_impl.cpp

```diff
@@ -151,6 +180,60 @@ void OffboardImpl::report_offboard_result(const Offboard::result_callback_t &cal
 
 void OffboardImpl::set_velocity_ned(Offboard::VelocityNEDYaw velocity_ned_yaw)
 {
+    _mutex.lock();
+    if (_mode != Mode::VELOCITY_NED) {
+        if (_call_every_cookie) {
+            // If we're already sending other setpoints, stop that now.
+            _parent->remove_call_every(_call_every_cookie);
+            _call_every_cookie = nullptr;
+        }
+        // We automatically send NED setpoints from now on.
+        _parent->add_call_every([this]() { send_velocity_ned(); },
+        SEND_INTERVAL_S,
+        &_call_every_cookie);
+
+        _mode = Mode::VELOCITY_NED;
+    } else {
+        // We're already sending these kind of setpoints. Since the setpoint change, let's
+        // reschedule the next call, so we don't send setpoints too often.
+        _parent->reset_call_every(_call_every_cookie);
+    }
+    _velocity_ned_yaw = velocity_ned_yaw;
```

Comment:
shouldn't we assign this before entering calling `send_velocity_ned()` at line:191.

Reply:
Good spotting! Call every will only schedule it and not run it immediately, however, it is still a race, so you're right!
Changed in 2c0a887.